The Qt5 version of QScintilla is called libqscintilla2-qt5.so.
Make it be referenced this way by Qt's build system.
--- Qt4Qt5/features/qscintilla2.prf.orig	2015-04-20 14:38:24 UTC
+++ Qt4Qt5/features/qscintilla2.prf
@@ -17,9 +17,17 @@ CONFIG(debug, debug|release) {
         win32: {
             LIBS += -lqscintilla2d
         } else {
-            LIBS += -lqscintilla2
+            greaterThan(QT_MAJOR_VERSION, 4) {
+                LIBS += -lqscintilla2-qt$${QT_MAJOR_VERSION}
+            } else {
+                LIBS += -lqscintilla2
+            }
         }
     }
 } else {
-    LIBS += -lqscintilla2
+    greaterThan(QT_MAJOR_VERSION, 4) {
+        LIBS += -lqscintilla2-qt$${QT_MAJOR_VERSION}
+    } else {
+        LIBS += -lqscintilla2
+    }
 }
